packages/ai/src/types.ts

```diff
@@ -25,53 +27,109 @@ export interface RAGRequest {
   stream?: boolean;
 }
 
-export interface MessageStart {
-  type: "message_start";
-  message: {
-    role: "assistant" | "system" | "user";
-    id: string;
-    model: string;
-  };
-}
+export type EdgeDBRagMessage =
+  | EdgeDBSystemMessage
+  | EdgeDBUserMessage
+  | EdgeDBAssistantMessage
+  | EdgeDBToolMessage;
 
-export interface ContentBlockStart {
-  type: "content_block_start";
-  index: number;
-  content_block: {
-    text: string;
-    type: "text";
-  };
+export interface EdgeDBSystemMessage {
+  role: "system";
+  content: string;
 }
 
-export interface ContentBlockDelta {
-  type: "content_block_delta";
-  delta: {
-    type: "text_delta";
-    text: string;
-  };
-  index: number;
+export interface EdgeDBUserMessage {
+  role: "user";
+  content: { type: "text"; text: string }[];
 }
 
-export interface ContentBlockStop {
-  type: "content_block_stop";
-  index: number;
+export interface EdgeDBAssistantMessage {
+  role: "assistant";
+  content: string;
+  tool_calls?: {
+    id: string;
+    type: "function";
+    function: { name: string; arguments: string };
+  }[];
 }
 
-export interface MessageDelta {
-  type: "message_delta";
-  delta: {
-    stop_reason: "stop";
-  };
+export interface EdgeDBToolMessage {
+  role: "tool";
+  content: string;
+  tool_call_id: string;
 }
 
-export interface MessageStop {
-  type: "message_stop";
-}
+export type StreamingMessage = z.infer<typeof _edgedbRagChunkSchema>;
 
-export type StreamingMessage =
-  | MessageStart
-  | ContentBlockStart
-  | ContentBlockDelta
-  | ContentBlockStop
-  | MessageDelta
-  | MessageStop;
+const _edgedbRagChunkSchema = z.discriminatedUnion("type", [
```

Comment:
I used this in the vercel provider so it was easier to copy paste it.

Reply:
Could we export it then maybe? Not a blocker.

Reply:
we can export it, I just dont understand why would someone need it? 
we export:
`export type StreamingMessage = z.infer<typeof _edgedbRagChunkSchema>;`
which will user most probably use?

Reply:
> I just dont understand/see when would someone need it?

Oh, I thought you mean that you are using a copy-pasted version of this schema in the Vercel provider, no? I'm saying export it here and import it in the Vercel provider instead of copy-pasting it. Or export it from the Vercel provider and import the type here. Whichever makes sense.

This is not a blocker for merging, it just feels a little strange here to have an essentially "private" Zod schema that isn't used for anything, but has to be kept in sync across two packages.

Reply:
hm okay I can do this, even tho it makes sense and is better to have one source of truth it feels a bit weird to me to install the whole lib just because one type, but maybe that's usually the way to go in this kind of situations?

Reply:
Oh, I hadn't noticed: Does the Vercel AI SDK Provider not use the `@edgedb/ai` package for anything already?